src/dawn/engine_dawn.h

```diff
@@ -58,6 +59,8 @@ class EngineDawn : public Engine {
 
  private:
   ::dawn::Device device_;
+  enum { kNumShaderTypes = unsigned(ShaderType::kTessellationEvaluation) + 1 };
+  ::dawn::ShaderModule shader_[kNumShaderTypes];
```

Comment:
Is this just because it's faster then map<ShaderType, ::dawn::ShaderModule> ?

Reply:
It hurts my head less.

Reply:
My concern is that we've now coupled this to ShaderType never changing order and never getting a new type added after TessellationEvaluation.